### PR TITLE
CAKE-5177 | Cleanup affiliate-slots service

### PR DESCRIPTION
**Pull request by vforge**
## Links

* http://wikia-inc.atlassian.net/browse/CAKE-5177

## Description

- Cleanup the affiliate-slots' logic, remove duplication
- Diff view makes it actually difficult to read

## Reviewers

@Wikia/cake

### CODE REVIEW

**Review by JoshuaRogan — APPROVED** [review]

File: app/services/affiliate-slots.js [hunk 1]
```diff
@@ -48,74 +86,93 @@ export default Service.extend({
...
+  getAllUnitsOnSearch(query) {
+    return this.getAllUnits([
+      t => !t.disableOnSearch,
+      t => checkFilter(t.query, query),
+    ]);
   },
 
   /**
-   * Get one random unit from that can be displayed on current wiki with given `query`
+   * Get all units that can be displayed on current wiki with given `title`
```
Comment (mklucsarits) [inline app/services/affiliate-slots.js]: This description is wrong; it gets only a single unit.

File: app/services/affiliate-slots.js [hunk 2]
```diff
@@ -5,11 +5,41 @@ import Service, { inject as service } from '@ember/service';
...
 
+/**
+  * @typedef {Object} AffiliateTargeting
+  * @property {string} [comment]
+  * @property {string[]} unit
+  * @property {number} [wikiId=[]]
+  * @property {number} [country=[]]
+  * @property {number|false} [page=[]]
+  * @property {number|false} [query=[]]
+  * @property {number} [vertical=[]]
```
Comment (mklucsarits) [inline app/services/affiliate-slots.js]: string, not number

File: app/services/affiliate-slots.js (hunk 1 above)
Comment (vforge) [reply app/services/affiliate-slots.js]: Fixed in 09a2066

File: app/services/affiliate-slots.js [hunk 3]
```diff
@@ -188,7 +188,7 @@ export default Service.extend({
       // grab only names of allowed units
       .map(t => t.unit)
       // flatten the array
-      .reduce((prev, curr) => prev.concat(curr))
+      .reduce((prev, curr) => prev.concat(curr), [])
```
Comment (JoshuaRogan) [inline app/services/affiliate-slots.js:191]: easy to forget 😓

File: app/services/affiliate-slots.js [hunk 4]
```diff
@@ -86,13 +86,27 @@ export default Service.extend({
   currentCountry: readOnly('geo.country'),
 
   /**
-   * Get one unit that can be displayed on search page with given `query`
+   * Get one BIG unit that can be displayed on search page with given `query`
+   *
+   * @param {string} query
+   * @returns {AffiliateUnit|undefined}
+   */
+  getBigUnitOnSearch(query) {
```
Comment (JoshuaRogan) [inline app/services/affiliate-slots.js:94]: (feel free to ignore) but maybe we should call it 'Fan Suggested' or something else other than 'Big Unit'

Comment (vforge) [reply app/services/affiliate-slots.js:94]: As this code is supposed (in the future) to handle different units I’d rather stick to big/small than to labels that the UI will have.

**Pull request merged**